plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -712,24 +712,10 @@ protected TestTable simpleTable()
         return new TestTable(onRemoteDatabase(), "tpch.simple_table", "(col BIGINT) Engine=Log", ImmutableList.of("1", "2"));
     }
 
-    @Test
     @Override
-    public void testCreateTableWithLongTableName()
+    protected void verifyTableNameLengthFailurePermissible(Throwable e)
     {
-        // Override because ClickHouse connector can create a table which can't be dropped
-        String baseTableName = "test_create_" + randomNameSuffix();
-        String validTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length());
-
-        assertUpdate("CREATE TABLE " + validTableName + " (a bigint)");
-        assertThat(getQueryRunner().tableExists(getSession(), validTableName)).isTrue();
-        assertThatThrownBy(() -> assertUpdate("DROP TABLE " + validTableName))
-                .hasMessageMatching("(?s).*(Bad path syntax|File name too long).*");
-
-        String invalidTableName = baseTableName + "z".repeat(maxTableNameLength().orElseThrow() - baseTableName.length() + 1);
-        assertThat(query("CREATE TABLE " + invalidTableName + " (a bigint)"))
-                .failure().hasMessageMatching("(?s).*(Cannot open file|File name too long).*");
-        // ClickHouse lefts a table even if the above statement failed
-        assertThat(getQueryRunner().tableExists(getSession(), validTableName)).isTrue();
+        assertThat(e).hasMessageMatching(".*The max length of table name for database tpch is [0-9]+, current length is [0-9]+.*\n");
```

Comment:
nit: The max length is static for specific version right ? Instead of regex - can we specify the length here ?

Reply:
> // The numeric value depends on file system

Despite it's not really a static one, we can reuse `maxTableNameLength()` here